src/main/java/engine/core/Engine.java

```diff
@@ -272,12 +273,17 @@ public void generateCPUScript(){
             checksForAttack.add(new Check("1", "1", Operator.EQUALS));
             ScriptCommand attack = new ScriptCommand(checksForAttack, Command.ATTACK);
             this.cpuCommands.add(attack);
+            
+            checksForDefense.add(new Check(Data.DISTANCE_FROM_OPPONENT.text(), "3",  Operator.GREATER_THAN_OR_EQUAL_TO));
```

Comment:
I would suggest moving 'before `checksForAttack`.  `checksForAttack` is always true since it is comparing "1" and "1" with the equality operator.  The `checksForDefense` portion will never actually be called.

Reply:
Alright will look into it.